client/app/components/game-client.js

```diff
@@ -0,0 +1,69 @@
+import Ember from 'ember';
+import GameChannel from 'vilive/utils/game-channel';
+
+export default Ember.Component.extend({
+  init() {
+    this.set('messages', Ember.makeArray());
+    GameChannel.init(this.get('session-manager').currentToken());
+    GameChannel.on('event', payload => {
+      this.putMessage(true, payload.body);
+    });
+    this._super(...arguments);
+  },
+
+  actions: {
+    sendCommand() {
+      const command = this.get('command');
+
+      if (!command || command.length === 0) { return; }
+
+      // hack for template at the moment
+      // false means client, true means server
+      this.putMessage(false, command);
+
+      GameChannel.push('user_cmd', {body: command});
+      this.set('command', null);
+    }
+  },
+
+  /**
+    This does not fire after each message is inserted just when the component
+    is inserted, in here we are going to focus on the text box.
+  */
+  didInsertElement() {
+    this.$('#command-input').focus();
+  },
+
+  /**
+    Need to scroll the div after the message has been rendered.
+    otherwise it stops short and scroll to last message inserted
+  */
+  scrollToBottomMessages() {
+    const div = this.$('.panel-body');
+    Ember.run.schedule('afterRender', () => {
+      div.scrollTop(div.prop("scrollHeight") - div.prop("clientHeight"));
+    });
+  },
+
+  /**
+    isServer drives the template since both client and server messages
+    are interlaced the template needs a way to know which is which.
+
+    unfortunately I can't use something like origin: "player" because
+    I'm not able to do someting like {{#if origin "player"}}
+
+    perhaps looking at writing a handle bar helper or something that makes
+    this cleaner or need a better way to template. I'm open to ideas.
+   */
+  putMessage(isServer, msg) {
+    this.get('messages').pushObject({
+      isServer: isServer,
+      body: msg
+    });
+    this.scrollToBottomMessages();
```

Comment:
moved `scrollTobottomMessages()` into here, for now. It should be called after each put message and the code really isn't putting several messages before calling render anyway.